File: FinalProject/e-commerce electronics/back-end/electronic/app/Models/OrderProduct.ts
import { DateTime } from 'luxon'
import { BaseModel, BelongsTo, belongsTo, column, ManyToMany, manyToMany } from '@ioc:Adonis/Lucid/Orm'
import Order from './Order'
import Product from './Product'
import Color from './Color'
import Size from './Size'

export default class OrderProduct extends BaseModel {
  public static table = 'order_products'

  @column({ isPrimary: true })
  public id: number

  @column({ serializeAs: 'order_id' })
  public orderId: number

  @column({ serializeAs: 'product_id' })
  public productId: number

  @column({ serializeAs: 'qty' })
  public qty: number

  @column({ serializeAs: 'price' })
  public price: number

  @column({ serializeAs: 'color_id' })
  public colorId: number | undefined

  @column({ serializeAs: 'size_id' })
  public sizeId: number | undefined

  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime

  @belongsTo(() => Order)
  public order: BelongsTo<typeof Order>

  @belongsTo(() => Product)
  public product: BelongsTo<typeof Product>

  @manyToMany(() => Color, {
    pivotTable: 'product_colors',
  })
  public colors: ManyToMany<typeof Color>;
  
  @manyToMany(() => Size, {
    pivotTable: 'product_sizes',
  })
  public sizes: ManyToMany<typeof Size>;
  

}
  
  
  

File: Loyalty-App/loyalty_app/lib/model/order_product.dart

import 'package:loyalty_app/model/product.dart';

class OrderProduct {
  int id;
  int orderId;
  int productId;
  int sizeId;
  int colorId;
  int? qty;
  double price;
  String createdAt;
  String updatedAt;
  Product product;

  OrderProduct({
    required this.id,
    required this.orderId,
    required this.productId,
    required this.sizeId,
    required this.colorId,
    required this.qty,
    required this.price,
    required this.createdAt,
    required this.updatedAt,
    required this.product,
  });

  Map<String, dynamic> toMap() {
    return {
      'orderId': orderId,
      'productId': productId,
      'sizeId': sizeId,
      'colorId': colorId,
      'qty': qty,
      'price': price,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
    };
  }

  factory OrderProduct.fromJson(Map<String, dynamic> json) {
  return OrderProduct(
    id: json['order_product_id'] as int? ?? 0,
    orderId: json['order_id'] as int? ?? 0,
    productId: json['product_id'] as int? ?? 0,
    sizeId: json['size_id'] as int? ?? 0,
    colorId: json['color_id'] as int? ?? 0,
    qty: json['qty'] as int?,
    price: (json['price'] as num?)?.toDouble() ?? 0.0,
    createdAt: json['created_at'] as String? ?? '',
    updatedAt: json['updated_at'] as String? ?? '',
    product: Product.fromJson(json['products']),
  );
}

}